boofuzz/fuzz_logger_csv.py

```diff
@@ -56,7 +56,7 @@ def log_recv(self, data):
         self._print_log_msg(["recv", len(data), self._format_raw_bytes(data), data.decode()])
 
     def log_send(self, data):
-        self._print_log_msg(["send", len(data), self._format_raw_bytes(data), data.decode()])
+        self._print_log_msg(["send", len(data), self._format_raw_bytes(data), repr(data)])
```

Comment:
@JsHuang What is the impact of this change?

Reply:
For none ascii data, data.decode() will throw exception

Reply:
I agree that in case we send non ascii data, the logger will raise an uncaught exception. Git blame shows that the decoding was added when we were working on Python 3 compatibility.
I'd even go further and change the `log_recv()` function too, just in case the target answers with non ascii formatted data.

I did some testing to see what the difference between those two methods are.
`decode()` will interpret all input including line breaks. The result of this is, that the scv file will have line breaks inside the data field. This seems to be allowed as long as the field is surrounded by quotation marks, which it currently is.
But then we have the problem that quotation marks inside the data stream will mess with this so I don't think it's a good idea to decode the data.
On the pro side, decoding the data does of course improve the readability greatly, but `repr()` seems to be the better choice to guarantee compatibility.

Reply:
Thanks all; makes sense.